packages/server/src/server.ts

```diff
@@ -770,10 +770,22 @@ const rpcApp = app.rpcApp();
 app.getKoaApp().use(async (ctx, next) => {
   await rpcApp(ctx, next);
   // Handle custom errors
+
+  // single request
   if (ctx.body && ctx.body.result instanceof JsonRpcError) {
     ctx.body.error = { ...ctx.body.result };
     delete ctx.body.result;
   }
+
+  // batch request
+  if (ctx.body && ctx.body instanceof Array) {
+    ctx.body.forEach((item) => {
+      if (item.result instanceof JsonRpcError) {
```

Comment:
Add a comment for the logic here

Reply:
Due to my refactor I am pretty sure sure that the whole of the KoaApp middleware is no longer needed, it used to be needed since there was a bug on the way the results/responses were built even when there were errors. but after the refactor and code improvements done on this ticket.  I don't think is needed.

So thanks for bringing this to my attention, I will remove the following code and nothing will break.
```
app.getKoaApp().use(async (ctx, next) => {
  await rpcApp(ctx, next);
  // Handle custom errors

  // single request
  if (ctx.body && ctx.body.result instanceof JsonRpcError) {
    ctx.body.error = { ...ctx.body.result };
    delete ctx.body.result;
  }

  // batch request
  if (ctx.body && ctx.body instanceof Array) {
    ctx.body.forEach((item) => {
      if (item.result instanceof JsonRpcError) {
        item.error = { ...item.result };
        delete item.result;
      }
    });
  }
});
```

Reply:
Okay, the KoaApp is needed 😅 
But the Handle of custom errors is not, this was a workaround due to another bug that was fixed during the refactors on this PR.

Reply:
Okay let's minimize the refactor where possible and open separate tickets if necessary.
if we improve as we go along though that's great